lib/ice_nine/freezer/object.rb

```diff
@@ -16,6 +16,8 @@ class Object < self
       #
       # @return [Object]
       def self.guarded_deep_freeze(object, recursion_guard)
+        return object unless object.respond_to?(:freeze)
```

Comment:
> I feel like undefing freeze is a pretty strong statement that it would not like to be frozen

That's a good point. I think it's fair to assume this as a convention. You can ignore my suggestion.